st2client/st2client/commands/triggerinstance.py

```diff
@@ -117,14 +118,24 @@ def run(self, args, **kwargs):
         if args.status:
             kwargs['status'] = args.status
 
-        return self.manager.query(limit=args.last, **kwargs)
+        result, count = self.manager.query(limit=args.last, **kwargs)
+        return (result, count)
 
     def run_and_print(self, args, **kwargs):
-        instances = self.run(args, **kwargs)
-        self.print_output(reversed(instances), table.MultiColumnTable,
-                          attributes=args.attr, widths=args.width,
-                          json=args.json, yaml=args.yaml,
-                          attribute_transform_functions=self.attribute_transform_functions)
+
+        instances, count = self.run(args, **kwargs)
+        if args.json or args.yaml:
+            self.print_output(reversed(instances), table.MultiColumnTable,
+                              attributes=args.attr, widths=args.width,
+                              json=args.json, yaml=args.yaml,
+                              attribute_transform_functions=self.attribute_transform_functions)
+        else:
+            self.print_output(reversed(instances), table.MultiColumnTable,
+                              attributes=args.attr, widths=args.width,
+                              attribute_transform_functions=self.attribute_transform_functions)
+            if args.last >= self.default_limit and count and int(count) > args.last:
```

Comment:
It looks like it doesn't correctly handle a scenario where you pass in `-n 1` for example but there are more results.

So this logic needs to be changed to handle that scenario as well.